### PR TITLE
Updated README and Index for limitations and policies.

### PR DESCRIPTION
**Pull request by EricPonvelle**
To address [OCM-2496](https://issues.redhat.com/browse/OCM-2496), I added a list of limitations for OCM Terraform provider along with an updated list of policies.

This PR also covers [OCM-2494](https://issues.redhat.com/browse/OCM-2494):
    image

### CODE REVIEW

File: docs/index.md [hunk 1]
```diff
@@ -56,6 +56,21 @@ Terraform creates and manages resources through application programming interfac
 
 For more information, see the [Terraform documentation](https://developer.hashicorp.com/terraform).
 
+## Limitations of OCM Terraform provider
+
+The following items are limitations with the current release of the OCM Terraform provider:
+
+* The latest version is not backward compatible with version 1.0.1.
+* When creating a cluster, the cluster uses AWS credentials configured on your local machine. These credentials provide access to the AWS API for validating your account.
+When creating a Node Pool, you need to specify your replica count. At least one of the following items should be defined to build the machine pool:
```
Comment (nirarg) [inline docs/index.md:65]: Missing another dot here

File: docs/index.md [hunk 2]
```diff
@@ -56,6 +56,21 @@ Terraform creates and manages resources through application programming interfac
...
+The following items are limitations with the current release of the OCM Terraform provider:
+
+* The latest version is not backward compatible with version 1.0.1.
+* When creating a cluster, the cluster uses AWS credentials configured on your local machine. These credentials provide access to the AWS API for validating your account.
+When creating a Node Pool, you need to specify your replica count. At least one of the following items should be defined to build the machine pool: 
+   * `replicas= "<count>"` 
+   * `min_replicas = "<count>"` 
+   * `max_replicas="<count>"` 
+   * `autoscaling_enabled=true`
+* The htpasswd identity provider does not support adding users or supporting multiple users.
```
Comment (nirarg) [inline docs/index.md:70]:
```suggestion
* The htpasswd identity provider does not support creating the identity provider with multiple users or adding additional users  to existing identity provider.
```

File: docs/index.md [hunk 3]
```diff
@@ -56,6 +56,21 @@ Terraform creates and manages resources through application programming interfac
...
+* Your S3 bucket that is created as part of the OIDC configuration must be created in the same region that you used when creating your cluster.
```
Comment (nirarg) [inline docs/index.md:71]:
```suggestion
* The S3 bucket that is created as part of the OIDC configuration must be created in the same region that you used when creating your cluster.
```

File: docs/index.md [hunk 4]
```diff
@@ -56,6 +56,21 @@ Terraform creates and manages resources through application programming interfac
...
+* You cannot generate the `operator_role_prefix` when creating your account roles.
```
Comment (nirarg) [inline docs/index.md:72]:
```suggestion
* The terraform provider doesn't support auto generated `operator_role_prefix`, you must provide it when creating the account roles.
```

File: README.md [hunk 5]
```diff
@@ -33,6 +33,21 @@ For more information about ROSA, see the Red Hat documentation [here](https://ac
...
+
+* The latest version is not backward compatible with version 1.0.1.
+* When creating a cluster, the cluster uses AWS credentials configured on your local machine. These credentials provide access to the AWS API for validating your account.
+When creating a Node Pool, you need to specify your replica count. At least one of the following items should be defined to build the machine pool: 
+   * `replicas= "<count>"` 
+   * `min_replicas = "<count>"` 
+   * `max_replicas="<count>"` 
+   * `autoscaling_enabled=true`
+* The htpasswd identity provider does not support adding users or supporting multiple users.
+* Your S3 bucket that is created as part of the OIDC configuration must be created in the same region that you used when creating your cluster.
```
[the remark this answers is not in the document]
Comment (amalykhi) [reply README.md:48]: Confirming.

File: README.md [hunk 6]
```diff
@@ -33,6 +33,21 @@ For more information about ROSA, see the Red Hat documentation [here](https://ac
...
+* You cannot generate the `operator_role_prefix` when creating your account roles.
```
[the remark this answers is not in the document]
Comment (amalykhi) [reply README.md:49]: I like Nir's suggestion

File: README.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (amalykhi) [reply README.md]: @nirarg Please, suggest how to write it? THe user is able to build the cluster in different region from his OIDC provider.
OIDC config, S3 bucket and OIDC provider should be created in the same region. Just build such cluster to double-confirm this.

**Pull request merged**